libavpipe/src/avpipe_xc.c

```diff
@@ -2048,10 +2049,12 @@ encode_frame(
                 2*encoder_context->calculated_frame_duration &&
             params->xc_type != xc_extract_images &&
             params->xc_type != xc_extract_all_images) {
-            elv_log("GAP detected, packet->pts=%"PRId64", video_encoder_prev_pts=%"PRId64", url=%s",
-                output_packet->pts, encoder_context->video_encoder_prev_pts, params->url);
-            encoder_context->forced_keyint_countdown -=
-                (output_packet->pts - encoder_context->video_encoder_prev_pts)/encoder_context->calculated_frame_duration - 1;
+
+            int fc = (output_packet->pts - encoder_context->video_encoder_prev_pts)/encoder_context->calculated_frame_duration - 1;
+            encoder_context->forced_keyint_countdown -= fc;
+
+            elv_log("GAP detected packet->pts=%"PRId64" video_encoder_prev_pts=%"PRId64" count=%d keying_count=%d url=%s",
+                output_packet->pts, encoder_context->video_encoder_prev_pts, fc, encoder_context->forced_keyint_countdown, params->url);
```

Comment:
Same logic - just move log down to log the frame count `fc`

Reply:
Feels like there should be an assertion that `calculated_frame_duration` is not zero